Not able to debug libvirt issues if launcher crashes

**What happened**:
A VM fails to start, the pod logs hint that it's a libvirt issues:

```
level=info timestamp=2018-07-13T09:22:14.125248Z pos=libvirt.go:256 component=virt-launcher msg="Connecting to libvirt daemon: qemu:///system"

level=error timestamp=2018-07-13T09:22:14.207290Z pos=libvirt_helper.go:146 component=virt-launcher msg="libvirtd exited, restarting"

level=error timestamp=2018-07-13T09:22:15.244135Z pos=libvirt_helper.go:146 component=virt-launcher msg="libvirtd exited, restarting"

level=error timestamp=2018-07-13T09:22:16.280373Z pos=libvirt_helper.go:146 component=virt-launcher msg="libvirtd exited, restarting"
…
panic: failed to connect to libvirtd: cannot connect to libvirt daemon: timed out waiting for the condition

goroutine 1 [running]:

main.createLibvirtConnection(0xc4200b4d80, 0x11f4c80)

    /root/go/src/kubevirt.io/kubevirt/cmd/virt-launcher/virt-launcher.go:111 +0x125

main.main()

    /root/go/src/kubevirt.io/kubevirt/cmd/virt-launcher/virt-launcher.go:253 +0x440

virt-launcher exited with code 2
```

When trying to exec into the compute pod, this fails, because the container continues to crash.

**What you expected to happen**:
Either get the libvirt logs on the pod logs. Or ability to log into the pod to debug libvirt

**How to reproduce it (as minimally and precisely as possible)**:
Make virt-launcher crash and try to debug libvirt

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`): 0.7.0-alpha.2
- Kubernetes version (use `kubectl version`): 1.11
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): ubuntu
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/cmd/virt-launcher/libvirtd.sh
+++ b/cmd/virt-launcher/libvirtd.sh
@@ -53,4 +53,7 @@ if [ -d "/dev/hugepages" ]; then
     echo 'hugetlbfs_mount = "/dev/hugepages"' >>/etc/libvirt/qemu.conf
 fi
 
+# Log to stderr
+echo "log_outputs = \"1:stderr\"" >> /etc/libvirt/libvirtd.conf
+
 /usr/sbin/libvirtd

--- a/pkg/virt-launcher/virtwrap/util/libvirt_helper.go
+++ b/pkg/virt-launcher/virtwrap/util/libvirt_helper.go
@@ -3,6 +3,7 @@ package util
 import (
 	"encoding/xml"
 	"fmt"
+	"os"
 	"os/exec"
 	"reflect"
 	"strings"
@@ -145,6 +146,10 @@ func StartLibvirt(stopChan chan struct{}) {
 		for {
 			exitChan := make(chan struct{})
 			cmd := exec.Command("/usr/share/kubevirt/virt-launcher/libvirtd.sh")
+
+			// libvirtd logs to stderr (see configuration in libvirtd.sh)
+			// connect libvirt's stderr to our own stdout in order to see the logs in the container logs
+			cmd.Stderr = os.Stdout
 
 			err := cmd.Start()
 			if err != nil {

--- a/tests/vmi_lifecycle_test.go
+++ b/tests/vmi_lifecycle_test.go
@@ -96,6 +96,23 @@ var _ = Describe("VMIlifecycle", func() {
 				Should(ContainSubstring("Found PID for"))
 		})
 
+		It("should log libvirtd logs", func() {
+			vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).To(BeNil())
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			By("Getting virt-launcher logs")
+			logs := func() string { return getVirtLauncherLogs(virtClient, vmi) }
+			Eventually(logs,
+				11*time.Second,
+				500*time.Millisecond).
+				Should(ContainSubstring("info : libvirt version: "))
+			Eventually(logs,
+				2*time.Second,
+				500*time.Millisecond).
+				Should(ContainSubstring("info : hostname: " + vmi.Name))
+		})
+
 		It("should reject POST if schema is invalid", func() {
 			jsonBytes, err := json.Marshal(vmi)
 			Expect(err).To(BeNil())

